single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/layout/MatrixZoneLayout.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.sld.layout;
+
+import com.powsybl.sld.layout.pathfinding.*;
+import com.powsybl.sld.layout.zonebygrid.*;
+import com.powsybl.sld.model.coordinate.*;
+import com.powsybl.sld.model.coordinate.Point;
+import com.powsybl.sld.model.graphs.*;
+import com.powsybl.sld.model.nodes.Node;
+import org.jgrapht.alg.util.*;
+
+import java.util.*;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+public class MatrixZoneLayout extends AbstractZoneLayout {
+    private final MatrixZoneLayoutModel model;
+    private final String[][] matrix;
+
+    private final PathFinder pathFinder;
+
+    protected MatrixZoneLayout(ZoneGraph graph, String[][] matrix, SubstationLayoutFactory sLayoutFactory, VoltageLevelLayoutFactory vLayoutFactory) {
+        super(graph, sLayoutFactory, vLayoutFactory);
+        this.model = new MatrixZoneLayoutModel(90);
+        this.matrix = matrix;
+        this.pathFinder = new PathFinderFactory().createDijkstra();
```

Comment:
I am wondering if we will have several PathFinder implementations. In this case, we should be able to customize the pathFinder (again, with a LayoutParameters parameter?)

Reply:
Good idea, this change can be made on an other PR if we add a new path finder algo